engine/consensus/matching/core.go

```diff
@@ -379,11 +379,24 @@ func (c *Core) ProcessFinalizedBlock(finalizedBlockID flow.Identifier) error {
 	// request execution receipts for unsealed finalized blocks
 	pendingReceiptRequests, firstMissingHeight, err := c.requestPendingReceipts()
 	requestReceiptsSpan.Finish()
-
 	if err != nil {
 		return fmt.Errorf("could not request pending block results: %w", err)
 	}
 
+	// Prune Execution Tree
+	_, seal, err := c.state.AtBlockID(finalizedBlockID).SealedResult()
+	if err != nil {
+		return fmt.Errorf("could not retrieve latest seal for the finalized block %v: %w", finalizedBlockID, err)
+	}
+	lastSealed, err := c.headersDB.ByBlockID(seal.BlockID)
+	if err != nil {
+		return fmt.Errorf("could not retrieve last sealed block %v: %w", seal.BlockID, err)
+	}
+	err = c.receipts.PruneUpToHeight(lastSealed.Height)
```

Comment:
execution receipts mempool also needs to be pruned by last sealed height, so we could remove those sealed receipts from mempool, which wasn't doing before.

Reply:
true, we could optimize this to 
```golang
c.receipts.PruneUpToHeight(lastSealed.Height+1)
```